### ISSUE TITLE
Installation issue: vtk depends on hdf5~hl which conflicts its dependency netcdf

### ISSUE THREAD

**Issue by ChrisTX** [spack/spack#11166]
Attempting to install `vtk` yields the following error:

```
[cpfeiffer@rhel7vm spack]$ ./bin/spack spec vtk
Input spec
--------------------------------
vtk

Concretized
--------------------------------
==> Error: An unsatisfiable variant constraint has been detected for spec:

    hdf5@1.10.5%gcc@8.2.1~cxx~debug~fortran~hl+mpi+pic+shared~szip~threadsafe arch=linux-rhel7-x86_64 


while trying to concretize the partial spec:

    netcdf@4.6.3%gcc@8.2.1~dap~hdf4 maxdims=1024 maxvars=8192 +mpi~parallel-netcdf+pic+shared arch=linux-rhel7-x86_64 
        ^m4@1.4.18%gcc@8.2.1+sigsegv arch=linux-rhel7-x86_64 
            ^libsigsegv
        ^mpi
        ^zlib@1.2.11%gcc@8.2.1+optimize+pic+shared arch=linux-rhel7-x86_64 


netcdf requires hdf5 variant +hl, but spec asked for ~hl
```

This is because `netcdf` requests `hdf5+hl` [here](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/netcdf/package.py#L106) and `vtk` depends on just `hdf5` [here](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/vtk/package.py#L82). Changing to `hdf5+hl` in the `vtk` packages resolves the issue.

**Comment by sethrj** [comment]
Same issue as https://github.com/spack/spack/issues/593 I believe

**Comment by flxmr** [comment]
https://github.com/spack/spack/pull/11297 fixed if this PR comes through

**Comment by chuckatkins** [comment]
You can work around this at the moment by adding `^hdf5+hl+mpi` to the vtk spec

**Comment by alalazo** [comment]
Duplicate of #1781

**Comment by alalazo** [comment]
This is solved by #19501
